### ISSUE TITLE
Expose blockTags in subParser 'hashHTMLBlock'

### ISSUE THREAD

**Issue by TheBorgel** [showdownjs/showdown#871]
I am trying to force showdown to ignore my custom HTML tags for my text editor. 
The block tags is a list of tags that will be ignored during parsing, and adding my custom tag names to this list achieves the desired result: i.e. the text inside the tags are not parsed. 

For now I achieved my desired result by adding a div tag around my custom tags like so:
```
type: 'lang',
regex: /<TagName(.*?)<\/TagName>/gs,
replace: '<div><TagName$1<\/TagName></div>'
```

**Comment by tivie** [comment]
In a way, the innards of hashHTMLBlock subparser is already exposed.

You can hook to the event `makehtml.hashHTMLBlocks.before` with a listener extension and replicate the functionality of the [HashHTMLBlock][1] parser yourself.


[1]: https://github.com/showdownjs/showdown/blob/master/src/subParsers/makehtml/hashHTMLBlocks.js#L60-L83

**Comment by TheBorgel** [comment]
True, but I would say that the extra effort for the user and the untidy end result is a bit unnecessary.

Would there not be a fairly common use case for being able to add or remove tags from the list to let the user customize the parser in a clean way?
